zcash_client_sqlite/src/wallet/sapling.rs

```diff
@@ -188,11 +219,19 @@ pub(crate) fn select_spendable_sapling_notes(
                  SUM(value)
                     OVER (PARTITION BY account, spent ORDER BY id_note) AS so_far
              FROM sapling_received_notes
-             INNER JOIN transactions ON transactions.id_tx = sapling_received_notes.tx
+             INNER JOIN transactions
+                ON transactions.id_tx = sapling_received_notes.tx
              WHERE account = :account
+             AND commitment_tree_position IS NOT NULL
              AND spent IS NULL
              AND transactions.block <= :anchor_height
              AND id_note NOT IN rarray(:exclude)
+             AND NOT EXISTS (
+                SELECT 1 FROM v_sapling_shard_unscanned_ranges unscanned
+                WHERE sapling_received_notes.commitment_tree_position >= unscanned.start_position
+                AND sapling_received_notes.commitment_tree_position < unscanned.end_position_exclusive
+                AND unscanned.block_range_start <= :anchor_height
```

Comment:
Same issue as above. Actually can we factor out the duplicated code?

Reply:
This does not have the same issue; this has the correct check.